### PR TITLE
test(*) add e2e CNI tests

### PR DESCRIPTION
**Pull request by nickolaev**
### Summary

Add CNI tests in e2e

### CODE REVIEW

**Comment by nickolaev** [comment]
@dependabot rebase

File: test/e2e/externalservices_kubernetes_test.go [hunk 1]
```diff
@@ -105,6 +105,7 @@ metadata:
 
 	var cluster Cluster
 	var clientPod *v1.Pod
+	var deployOptsFuncs []DeployOptionsFunc
```
Comment (jakubdyszkiewicz) [inline test/e2e/externalservices_kubernetes_test.go:108]: are changes in this file required or can we revert it?

File: test/e2e/externalservices_universal_test.go [hunk 2]
```diff
@@ -69,6 +69,7 @@ networking:
 	es2 := "2"
 
 	var cluster Cluster
+	var deployOptsFuncs []DeployOptionsFunc
```
Comment (jakubdyszkiewicz) [inline test/e2e/externalservices_universal_test.go:72]: are changes in this file required or can we revert it?

File: test/e2e/kuma_deploy_global_remote_test.go [hunk 3]
```diff
@@ -36,27 +36,33 @@ metadata:
...
-			Verbose)
+			Silent)
 		Expect(err).ToNot(HaveOccurred())
 
 		c1 = clusters.GetCluster(Kuma1)
-		c2 = clusters.GetCluster(Kuma2)
 
 		err = NewClusterSetup().
-			Install(Kuma(core.Global)).
+			Install(Kuma(core.Global, optsGlobal...)).
```
Comment (jakubdyszkiewicz) [inline test/e2e/kuma_deploy_global_remote_test.go:51]: is this required?

File: test/framework/k8s_cluster.go [hunk 4]
```diff
@@ -654,3 +653,31 @@ func (c *K8sCluster) Deploy(deployment Deployment) error {
...
+		DefaultTimeout)
+
+	pods := k8s.ListPods(c.t,
+		c.GetKubectlOptions(namespace),
+		metav1.ListOptions{
+			LabelSelector: "app=" + name,
+		},
+	)
+	if len(pods) < 1 {
+		return errors.Errorf("Kuma CP pods: %d", len(pods))
```
Comment (jakubdyszkiewicz) [inline test/framework/k8s_cluster.go:674]: I don't think it's always Kuma CP anymore
```
return errors.Errorf("%s pods: %d", name, len(pods))
```

File: test/e2e/externalservices_kubernetes_test.go (hunk 1 above)
Comment (nickolaev) [reply test/e2e/externalservices_kubernetes_test.go:108]: Let's keep these for uniformity of tests

**Review by jakubdyszkiewicz — APPROVED** [review]

**Pull request merged**